### PR TITLE
Financial Connections: added support for reduced_branding

### PR DESCRIPTION
**Pull request by kgaidis-stripe**
## Summary

^ adds support for `reduced_branding` flag which hides logo in navigation bar

Links for context:
- https://stripe.slack.com/archives/C02KW8G938W/p1670955745288589?thread_ts=1670955422.450539&cid=C02KW8G938W
- https://jira.corp.stripe.com/browse/BANKCON-5899
- https://jira.corp.stripe.com/browse/BANKCON-5586
- https://www.figma.com/file/pU6S60e08Rgv0dvycCxMT8/Connections---Mobile?node-id=2884%3A52462&t=rzvAXP5ys05asf6D-0
- https://git.corp.stripe.com/stripe-internal/stripe-js-v3/pull/13083
- https://github.com/stripe/stripe-android/pull/5949
- https://amp.corp.stripe.com/feature-flags/flag/bank_connections_enable_visual_features_in_synchronize_method/rollouts/ffrp_N0wFXcGgwweAOc

## Testing

Screen Shot 2022-12-20 at 2 27 47 PM

# Logo Visible (BEFORE)

https://user-images.githubusercontent.com/105514761/208525216-935fd81c-5259-4f7f-a903-e9a6f56c7e34.mov

# Logo NOT Visible (AFTER)

https://user-images.githubusercontent.com/105514761/208525246-9cb86098-95c5-4774-8e1e-51c859215e18.mov

### CODE REVIEW

File: StripeFinancialConnections/StripeFinancialConnections/Source/API Bindings/Models/FinancialConnectionsSynchronize.swift
Comment (kgaidis-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/API Bindings/Models/FinancialConnectionsSynchronize.swift:14]: Logic double-check:

https://livegrep.corp.stripe.com/view/stripe-internal/pay-server/lib/bank_connections/api/data/client_update.rb#L18

https://livegrep.corp.stripe.com/view/stripe-internal/pay-server/lib/bank_connections/api/data/visual_update.rb#L9

**Comment by kgaidis-stripe** [comment]
@vardges-stripe ready for review; the one out-standing thing I will do before merging is ensure the payload from server is parsed correctly (waiting for backend to land the flag)

**Review by vardges-stripe — APPROVED** [review]
nice

**Pull request merged**
